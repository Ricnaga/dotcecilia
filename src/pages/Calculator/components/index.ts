export * from './Agreement/Agreement';
export * from './CalculatorTable/CalculatorTable';
export * from './Discount/Discount';
export * from './ExtraHour/ExtraHour';
export * from './Unsanitary/Unsanitary';
