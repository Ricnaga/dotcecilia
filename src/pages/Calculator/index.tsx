import {
  CalculatorDiscount,
  CalculatorExtraHour,
  CalculatorTransport,
} from './components';

export function CalculatorPage() {
  return (
    <div className="w-full mt-8">
      <p className="text-5xl font-bold mb-12">Calculadora</p>

      <div className="grid grid-cols-12 gap-10">
        <div className="grid grid-cols-12 col-span-7">
          <div className="col-span-12 mb-4">
            <CalculatorExtraHour />
          </div>
          <div className="col-span-12 mb-4">
            <CalculatorDiscount />
          </div>
          <div className="col-span-12 mb-4">
            <CalculatorTransport />
          </div>
        </div>
        <div className="col-span-5">
          <p className="text-4xl text-center font-bold mb-4">Total</p>
        </div>
      </div>
    </div>
  );
}
