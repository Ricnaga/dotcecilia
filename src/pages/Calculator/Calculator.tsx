import { PageContainer } from '@shared/components';
import { Accordion } from '@shared/components/Accordion/Accordion';
import { Agreement, Discount, ExtraHour, Unsanitary } from './components';

export function Calculator() {
  return (
    <PageContainer title="Calculadora">
      <div className="grid grid-cols-3 gap-4">
        <div className="col-span-2 grid grid-cols-2 gap-4">
          <div className="col-span-2">
            <Accordion title="Hora extra">
              <ExtraHour />
            </Accordion>
          </div>
          <div>
            <Accordion title="Acerto">
              <Agreement />
            </Accordion>
          </div>
          <div>
            <Accordion title="Desconto">
              <Discount />
            </Accordion>
          </div>
        </div>
        <div>
          <Accordion title="Insalubridade">
            <Unsanitary />
          </Accordion>
        </div>
      </div>
    </PageContainer>
  );
}
