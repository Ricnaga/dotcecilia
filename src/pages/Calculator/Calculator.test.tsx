import { render } from '@testing-library/react';
import { Calculator } from './Calculator';

describe('Page: Calculator', () => {
  it('should render correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
