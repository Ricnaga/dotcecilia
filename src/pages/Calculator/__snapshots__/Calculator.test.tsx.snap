// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Calculator should render correctly 1`] = `
<div>
  <div>
    <p
      class="text-5xl font-bold mt-8 mb-6"
    >
      Calculadora
    </p>
    <div
      class="grid grid-cols-3 gap-4"
    >
      <div
        class="col-span-2 grid grid-cols-2 gap-4"
      >
        <div
          class="col-span-2"
        >
          <details
            class="accordion-container group"
          >
            <summary
              class="accordion-summary"
            >
              <p
                class="flex flex-grow"
              >
                Hora extra
              </p>
              <svg
                aria-hidden="true"
                class="group-open:animate-rotate-180 group-open:rotate-180 animate-rotate-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </summary>
            <div
              class="accordion-core group-open:animate-slide"
            >
              <div
                class="grid grid-cols-3 gap-4"
              >
                <div
                  class="col-1 flex flex-col gap-2 justify-around"
                >
                  <div>
                    <div
                      data-error="false"
                      data-variants="input-wrapper"
                    >
                      <input
                        data-variants="input"
                        placeholder="Salário"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div>
                    <div
                      data-error="false"
                      data-variants="input-wrapper"
                    >
                      <input
                        data-variants="input"
                        placeholder="Horas trabalhadas"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="col-span-2 grid grid-cols-2 gap-4"
                >
                  <table
                    class="bg-teal-600 text-gray-100 rounded-xl hover:brightness-105 transition-all"
                  >
                    <thead>
                      <tr>
                        <th
                          class="bg-teal-700 rounded-t-xl text-xlg font-bold leading-5 text-center p-2"
                          colspan="2"
                        >
                          Salário
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          Valor/hora
                        </td>
                        <td
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          6% VTR
                        </td>
                        <td
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          Adiantamento
                        </td>
                        <td
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          Desconto dias faltados
                        </td>
                        <td
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  <table
                    class="bg-teal-600 text-gray-100 rounded-xl hover:brightness-105 transition-all"
                  >
                    <thead>
                      <tr>
                        <th
                          class="bg-teal-700 rounded-t-xl text-xlg font-bold leading-5 text-center p-2"
                          colspan="2"
                        >
                          Hora extra
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          60 %
                        </td>
                        <td
                          align="right"
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                      <tr
                        class="border-t"
                      >
                        <td
                          class="px-6 py-1"
                        >
                          100 %
                        </td>
                        <td
                          align="right"
                          class="px-6 py-1"
                        >
                          R$ 0,00
                        </td>
                      </tr>
                    </tbody>
                    <tfoot>
                      <tr>
                        <td
                          align="center"
                          class="p-2 border-t"
                          colspan="2"
                        >
                          Total: 
                          R$ 0,00
                        </td>
                      </tr>
                    </tfoot>
                  </table>
                </div>
              </div>
            </div>
          </details>
        </div>
        <div>
          <details
            class="accordion-container group"
          >
            <summary
              class="accordion-summary"
            >
              <p
                class="flex flex-grow"
              >
                Acerto
              </p>
              <svg
                aria-hidden="true"
                class="group-open:animate-rotate-180 group-open:rotate-180 animate-rotate-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </summary>
            <div
              class="accordion-core group-open:animate-slide"
            >
              <div
                class="grid grid-cols-2 gap-4"
              >
                <div
                  class="flex flex-col gap-2 justify-around"
                >
                  <div>
                    <div
                      data-error="false"
                      data-variants="input-wrapper"
                    >
                      <input
                        data-variants="input"
                        placeholder="Salário"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div>
                    <div
                      data-error="false"
                      data-variants="input-wrapper"
                    >
                      <input
                        data-variants="input"
                        placeholder="Meses trabalhados"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <table
                  class="bg-teal-600 text-gray-100 rounded-xl hover:brightness-105 transition-all"
                >
                  <thead>
                    <tr>
                      <th
                        class="bg-teal-700 rounded-t-xl text-xlg font-bold leading-5 text-center p-2"
                        colspan="2"
                      >
                        Salário
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class="border-t"
                    >
                      <td
                        class="px-6 py-1"
                      >
                        Férias
                      </td>
                      <td
                        class="px-6 py-1"
                      >
                        R$ 0,00
                      </td>
                    </tr>
                    <tr
                      class="border-t"
                    >
                      <td
                        class="px-6 py-1"
                      >
                        13º
                      </td>
                      <td
                        class="px-6 py-1"
                      >
                        R$ 0,00
                      </td>
                    </tr>
                    <tr
                      class="border-t"
                    >
                      <td
                        class="px-6 py-1"
                      >
                        Férias - 1/3
                      </td>
                      <td
                        class="px-6 py-1"
                      >
                        R$ 0,00
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </details>
        </div>
        <div>
          <details
            class="accordion-container group"
          >
            <summary
              class="accordion-summary"
            >
              <p
                class="flex flex-grow"
              >
                Desconto
              </p>
              <svg
                aria-hidden="true"
                class="group-open:animate-rotate-180 group-open:rotate-180 animate-rotate-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </summary>
            <div
              class="accordion-core group-open:animate-slide"
            >
              <div
                class="grid grid-cols-2 gap-4"
              >
                <div
                  class="flex flex-col gap-2 justify-around"
                >
                  <div>
                    <div
                      data-error="false"
                      data-variants="input-wrapper"
                    >
                      <input
                        data-variants="input"
                        placeholder="Dias faltados"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <table
                  class="bg-teal-600 text-gray-100 rounded-xl hover:brightness-105 transition-all"
                >
                  <thead>
                    <tr>
                      <th
                        class="bg-teal-700 rounded-t-xl text-xlg font-bold leading-5 text-center p-2"
                        colspan="2"
                      >
                        Salário
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class="border-t"
                    >
                      <td
                        class="px-6 py-1"
                      >
                        Dias faltados
                      </td>
                      <td
                        class="px-6 py-1"
                      >
                        0
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </details>
        </div>
      </div>
      <div>
        <details
          class="accordion-container group"
        >
          <summary
            class="accordion-summary"
          >
            <p
              class="flex flex-grow"
            >
              Insalubridade
            </p>
            <svg
              aria-hidden="true"
              class="group-open:animate-rotate-180 group-open:rotate-180 animate-rotate-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </summary>
          <div
            class="accordion-core group-open:animate-slide"
          >
            <div
              class="grid grid-cols-2 gap-4 items-center"
            >
              <div>
                <div
                  data-error="false"
                  data-variants="input-wrapper"
                >
                  <input
                    data-variants="input"
                    placeholder="Salário"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <table
                class="bg-teal-600 text-gray-100 rounded-xl hover:brightness-105 transition-all"
              >
                <thead>
                  <tr>
                    <th
                      class="bg-teal-700 rounded-t-xl text-xlg font-bold leading-5 text-center p-2"
                      colspan="2"
                    >
                      Valor
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    class="border-t"
                  >
                    <td
                      class="px-6 py-1"
                    >
                      Insalubridade
                    </td>
                    <td
                      class="px-6 py-1"
                    >
                      R$ 264,00
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;
